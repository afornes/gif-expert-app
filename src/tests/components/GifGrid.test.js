import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import GifGrid from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Tests for Gif Grid', () => { 
    
    const category = 'Kimetsu no yaiba';
    
    test('should show gif grid', () => { 
        
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        
        const wrapper = shallow(<GifGrid category={ category }/>);
        expect(wrapper).toMatchSnapshot();
    });
    
    test('should show gifs when useFetchGifs received', () => { 
        
        const gifs = [
            {
                id: 'ABC',
                url: 'https://localhost/cosas.gif',
                title: 'Cosas'
            },
            {
                id: 'DEF',
                url: 'https://localhost/cosas-2.gif',
                title: 'MÃ¡s Cosas'
            }
        ];
        
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        
        const wrapper = shallow(<GifGrid category={ category }/>);
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
        
    });
});