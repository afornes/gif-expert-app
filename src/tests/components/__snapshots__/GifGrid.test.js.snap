// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Gif Grid should show gif grid 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Kimetsu no yaiba
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
